jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -605,6 +606,58 @@ def vue_set_value(self, data):
         value = data.get('value')
         setattr(self, attr_name, value)
 
+    def apply_RGB_presets(self):
+        """
+        Applies preset colors, opacities, and stretch settings to all visible layers
+        (in all viewers) when in Monochromatic mode.
+        """
+
+        if (self.image_color_mode_value != "One color per layer" or
+                self.image_color_mode_sync['mixed']):
+            raise ValueError("RGB presets can only be applied if color mode is Monochromatic.")
+        # Preselected colors we want to use for 5 or less layers
+        preset_colors = [self.swatches_palette[0][0],
+                         "#0000FF",
+                         "#00FF00",
+                         self.swatches_palette[1][0],
+                         self.swatches_palette[4][1]]
+
+        # Switch back to this at the end
+        initial_layer = self.layer_selected
+
+        # Filter out subset layers
+        image_layers = [layer for layer in self.layer.choices if "Subset" not in layer]
+        visible_layers = []
+        for layer in image_layers:
+            self.layer_selected = layer
+            if self.image_visible.value:
+                visible_layers.append(layer)
+
+        # Set opacity to something that seems sensible
+        n_visible = len(visible_layers)
+        default_opacity = 1
+        if n_visible > 2:
+            default_opacity = 1 / math.log2(n_visible)
+        # Sample along a colormap if we have too many layers
+        if n_visible > 5:
+            cmap = matplotlib.colormaps['gist_rainbow'].resampled(n_visible)

```

Comment:
I'm a bit worried about two different if-else statements that need to be synced to each other (if we ever change the length of the supported number of presets, we need to update the list above, this if-else, _and_ the if-else in the for loop.  Maybe we can create the list of colors before the for-loop and then just access and set them within the for loop?

Reply:
Yeah, I don't love all the conditionals here either. Creating a matplotlib colormap is cheap enough I think I'll just do it anytime there are more than 2 visible layers in that first if statement. Actually, instead of hard-coding numbers, I can check against the length of `preset_colors`.

Reply:
what if we had (before the for-loop): 

```
if n_visible > len(preset_colors):
    preset_colors = [matplotlib.colors.to_hex(cmap(i), keep_alpha=True) for i in range(n_visible)]
```

and then in the for-loop, just access `preset_colors[i]` (or better yet, loop over preset colors as well)?

Reply:
Can't quite just loop over the preset colors, since we also need to get the right index from the layer list in the loop. I made the other changes though.

Reply:
Could loop over both with zip (`for visible_layer, preset_color in zip(visible_layers, preset_colors)`) if we don't need to access much else, but this works too